index.js

```diff
@@ -12,6 +12,23 @@ module.exports = (robot, _, Settings = require('./lib/settings')) => {
 
     if (defaultBranch && settingsModified) {
       return Settings.sync(context.github, context.repo())
+    } else if (!defaultBranch && settingsModified) {
+      let settingChangeCommits = payload.commits.filter(commit => { // get all commits that change settings.yml
```

Comment:
You really only need to set the status on the latest commit in the push. You can do every single one, but most integrations don't bother since GitHub only uses the last one to determine merge-ability.